pip_audit/_cli.py

```diff
@@ -327,6 +327,12 @@ def _parser() -> argparse.ArgumentParser:  # pragma: no cover
             "this option can be used multiple times"
         ),
     )
+    parser.add_argument(
+        "--with-pip-compile",
+        action="store_true",
+        help="don't perform dependency installation in a temporary virtual env"
+        " if requirements files is builded with pip compile; requires --requirement argument",
+    )
```

Comment:
This implies that hashed requirements files always come from `pip-compile`, which isn't true. 

Rather than adding a new flag here, let's revert the behavior that was removed with #540: the pre-existing `--no-deps` and `--require-hashes` flags should cause us to go down an optimized path.